src/main/java/org/spongepowered/api/text/TextFactory.java

```diff
@@ -22,71 +22,48 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.text.message;
-
-import org.spongepowered.api.text.translation.Translation;
+package org.spongepowered.api.text;
 
 /**
  * Represents the required implementation for the static methods in
- * {@link Messages}.
+ * {@link Texts}.
  */
-interface MessageFactory {
+public interface TextFactory {
 
     /**
-     * Creates a {@link MessageBuilder} with empty text.
+     * Parses the specified JSON text and returns the parsed result.
      *
-     * @return A new message builder with empty text
+     * @param json The valid JSON text
+     * @return The parsed text
+     * @throws IllegalArgumentException If the JSON is invalid
      */
-    MessageBuilder createEmptyBuilder();
+    Text parseJson(String json) throws IllegalArgumentException;
 
     /**
-     * Creates a {@link MessageBuilder.Text} with the specified text.
+     * Parses the specified JSON text leniently and returns the parsed result.
      *
-     * @param text The text for the message
-     * @return A new message builder with the specified text
-     * @see Message.Text
+     * @param json The JSON text
+     * @return The parsed text
+     * @throws IllegalArgumentException If the JSON couldn't be parsed
      */
-    MessageBuilder.Text createTextBuilder(String text);
+    Text parseLenientJson(String json) throws IllegalArgumentException;
```

Comment:
Should be `parseJsonLenient` not `parseLenientJson`. The json we're parsing is not lenient, the way we're parsing it is.